### DISCUSSION TITLE
device type - predefined interface with device-name placeholer

### DISCUSSION THREAD

**Discussion by Frank17489** [netbox-community/netbox#16940] [Q&A] [answered]
Hi,

I would create a device-type with interfaces and ports.
Is it possible to defined an interfacename with a placeholder for the  devicename (with I create later) ?

perhaps **{{ devicename}}.lan** - I try this, but it dosn't work.

Its used to find quick the right interface in selectboxes ( instead of 1000 lan or eth entries)

thanks for any ideas

**Comment by AADIP** [comment]
I don't know of a way to do this inside of interface templates as you described. However, you could implement a custom script and event rule combination on creation of a device to populate the interface(s). It should be a very simple and small script to accomplish what you're looking for.

**Reply by Frank17489** [reply]
ok, it's an idea. my skills are not yet so developed, but I try this. thx

**Reply by Frank17489** [reply]
Hi AADIP,
In this case I can work with event rule. now its ready and works. 

thanks

**Comment by candlerb** [comment]
> Its used to find quick the right interface in selectboxes ( instead of 1000 lan or eth entries)

Which Netbox version are you using?

It sounds like you are working around a perceived weakness in the GUI. If you describe exactly what you're trying to do that shows you "1000 lan or eth entries" then we may be able to show a better way.

In most cases you can select the device first, and then you'll be presented with only the interfaces of that device.  In some versions of Netbox you need to click a little icon that looks like a cylinder to do this.

**Reply by Frank17489** [reply]
I use the netbox v4.0.5. 
I can use 2 ways to add a IP address to an interface.
1. I select the interface and add an IP 
    1.1. create a new IP -  in this formular I can't identifiy a free IP - ( first I must check in IP Ranges for a free IP address )
    or 1.2 assign IP - normally netbox has no free IPs, If a device has been deleted, the IP is also deleted
 2. In IP ranges I identify free IP addresses, create this and would not go back to the interface side. If a try to add the interface in the IP formular, I see 1000 of lan interfaces, no chance to select the device and than the interface.

**Reply by candlerb** [reply, marked as answer]
> If a try to add the interface in the IP formular, I see 1000 of lan interfaces, no chance to select the device and than the interface.

Do you mean you are looking at, for example, `/ipam/ip-addresses/516/edit/` and then trying to assign it to an interface?

![image](https://github.com/user-attachments/assets/b255dcbf-f713-467c-a213-9c968c89a627)

First, click on the little icon to the right

![image](https://github.com/user-attachments/assets/cab26108-b04b-43b4-9fc1-3f69b9cf2e46)

Then select a device from the Device dropdown (filter on site or rack first if necessary to reduce the list of devices):

![image](https://github.com/user-attachments/assets/a93a4144-3613-49f9-8654-9a4a26e50a88)

Then you will see only the interfaces on that device:

![image](https://github.com/user-attachments/assets/0b3754b4-883d-41c2-a575-bf1374db8229)

Then click on the interface you want in that list, e.g. "g2"

Admittedly this UI is not great, and many people (including me) think it is worse than how it was in older versions of Netbox. But it's serviceable once you know how.

However, I think this is much better than giving fake names to interfaces where they include the device name. Doing so would make Netbox *much* worse as a source of truth, as it would be difficult (for example) to generate configurations from templates, when all the interface names are wrong.

**Reply by Frank17489** [reply]
Oh, this possible filter have I overseen. thanks
